sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -39,11 +40,44 @@ def test_n_features_to_select(direction, n_features_to_select):
     sfs.fit(X, y)
     if n_features_to_select is None:
         n_features_to_select = 5  # n_features // 2
+
     assert sfs.get_support(indices=True).shape[0] == n_features_to_select
     assert sfs.n_features_to_select_ == n_features_to_select
     assert sfs.transform(X).shape[1] == n_features_to_select
 
 
+@pytest.mark.parametrize('direction', ('forward', 'backward'))
+def test_stopping_criterion(direction):
+    # Make sure n_features_to_select is respected
+
+    X, y = make_regression(n_features=10)
```

Comment:
Maybe add a large fraction of non-informative features with `n_features=50` and `n_informative=10` to make the impact of feature selection more visible.

Reply:
Also please pass `random_state=0` to make this test deterministic.

Reply:
fix it on https://github.com/scikit-learn/scikit-learn/pull/20145/commits/b5a332eb2f87b74ec9a97f13a6fdb116372a0fa2